Algorithms/Problems/NQueenProblem/NQueenProblem.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Linq;
+
+namespace Algorithms.Problems
+{
+    public static class NQueenProblem
+    {
+        /// <summary>
+        /// Solves N-Queen Problem given a n dimension chessboard and using backtracking with recursion algorithm.
+        /// If we find a dead-end within or current solution we go back and try another position for queen.
+        /// </summary>
+        /// <param name="board">initial board.</param>
+        /// <param name="n">chessboard dimensions.</param>
+        /// <param name="col">starting column.</param>
+        /// <returns>Matrix solved. </returns>
+        public static bool BacktrackSolve(int[,] board, int n, int col)
```

Comment:
Really need to make a facade for it, consumers don't have to pass the 0 to the col parameter. n doesn't need to be specified either because it's contained in the board's dimensions.

Reply:
Could you make it return all possible solutions lazily as IEnumerable<int[,]>?